packages/react-router/src/link.tsx

```diff
@@ -644,30 +713,32 @@ export type LinkProps<
     | ((state: { isActive: boolean }) => React.ReactNode)
 }
 
+type LinkComponentProps<TComp> = React.PropsWithoutRef<
```

Comment:
This is nice. Basically because `TComp` doesn't change on every `Link` we can seperate these props into a seperate type and TS will cache this for all `Link`'s